internal/pipe/aur/aur.go

```diff
@@ -0,0 +1,477 @@
+package aur
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/apex/log"
+	"github.com/goreleaser/goreleaser/internal/artifact"
+	"github.com/goreleaser/goreleaser/internal/client"
+	"github.com/goreleaser/goreleaser/internal/commitauthor"
+	"github.com/goreleaser/goreleaser/internal/git"
+	"github.com/goreleaser/goreleaser/internal/pipe"
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/config"
+	"github.com/goreleaser/goreleaser/pkg/context"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	pkgBuildExtra     = "AURConfig"
+	defaultSSHCommand = "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i {{ .KeyPath }} -F /dev/null"
+	defaultCommitMsg  = "Update to {{ .Tag }}"
+)
+
+var ErrNoArchivesFound = errors.New("no linux archives found")
+
+// Pipe for arch linux's AUR pkgbuild.
+type Pipe struct{}
+
+func (Pipe) String() string                 { return "arch user repositories" }
+func (Pipe) Skip(ctx *context.Context) bool { return len(ctx.Config.AURs) == 0 }
+
+func (Pipe) Default(ctx *context.Context) error {
+	for i := range ctx.Config.AURs {
+		pkg := &ctx.Config.AURs[i]
+
+		pkg.CommitAuthor = commitauthor.Default(pkg.CommitAuthor)
+		if pkg.CommitMessageTemplate == "" {
+			pkg.CommitMessageTemplate = defaultCommitMsg
+		}
+		if pkg.Name == "" {
+			pkg.Name = ctx.Config.ProjectName + "-bin"
+			if len(pkg.Conflicts) == 0 {
```

Comment:
I would move this logic for  `Conflicts` and `Provides` out of the `pkg.Name` if clause and unconditionally check if the user have included `ProjectName` in the arrays. This would probably help produce more correct `PKGBUILD` files as people don't need to (and probably don't) realize that these are required.

Reply:
good call ,fixed

Reply:
Sorry, I was imprecise. You need to also check if `pkg.Name != ctx.Config.ProjectName` and if that is the case do the provides and conflicts. This is implied in the if-clause but there won't be any guards against this if `pkg.Name` is defined.